Library/Homebrew/extend/os/mac/formula_cellar_checks.rb

```diff
@@ -6,7 +6,7 @@ def check_shadowed_headers
       formula.name.start_with?(formula_name)
     end
 
-    return if formula.name =~ /^php\d+$/
+    return if formula.name =~ /^php(?:\@\d\.\d|\d+)?$/
```

Comment:
So the at sign doesn't need to be escaped the ?
That regexp does not seem to match php71 which is the current filenames used in the php tap. We could do /^php(\d+|@\d+\.\d+)?$/ which would also match 7.10 if the version would be written with double digits.  I think your regexp was also designed to match  php@7 as well as php@7.0 but other Formulas like ruby write the .0 so I'm not sure if we need to cover that case ?